contracts/exchange/Erc20Escrow.sol

```diff
@@ -94,6 +94,9 @@ contract Erc20Escrow is IErc20Escrow, EscrowBase {
 
         escrowedByOrder[_orderId].amount = escrowedByOrder[_orderId].amount - _amount;
         IERC20Upgradeable(escrowedByOrder[_orderId].token).transfer(_receiver, _amount);
+        if (escrowedByOrder[_orderId].amount == 0) {
```

Comment:
I don't like this because you're only doing the deleting the token address for withdraw, but not withdrawBatch().

Realistically, I think that to make it way cheaper and remove "token" redundancy, it's better to change escrowedByOrder from mapping(orderId => EscrowAmount) to:
**mapping(tokenAddress => mapping(orderId => amount))** 

this way, instead of storing a whole struct {token, amount}, it uses token as an index and it only stores {amount}. 

You'll have to update all of erc20Escrow.sol though for every escrowByOrder though

Reply:
Ah, nevermind with this. I realized why I created the structure in the first place.

Reply:
If you're going to delete the 'token', you might as well just delete the whole struct escrowedByOrder[orderId] instead of just deleting the token.

Make sure you delete it in withdrawBatch() as well. And check by how much this actually saves.

Reply:
This is the same deal as in NftEscrow, deleting the whole struct just means setting  .token to the zero address and .amount to zero (which is already zero), so it saves more gas to just delete the token specifically.

withdrawBatch is tricky because it runs the loop of updating the mapping and deleting the variables before sending a the transfer function at the end. And you can't transfer the token if you've already deleted the the tokenAddress. And according to the tests, declaring a local address variable costs quite a bit more than the amount refunded by deleting the tokenAddresses- UNLESS you have a long list of orderIds. So most of the time, it's not worth it.

Reply:
If that's the case, then you shouldn't delete just in the single Withdraw function. The gas savings is not worth the inconsistency between withdraw and withdrawBatch. The maintainability will go down because people will continually ask in the future why something is done in withdraw, but not withdraw batch. The inconsistency may also create future bugs because something gets deleted in withdraw but not withdrawBatch and it won't be common knowledge.